test/DynamoCoreTests/DSEvaluationViewModelTest.cs

```diff
@@ -159,13 +159,41 @@ private void SelectivelyAssertValues(MirrorData data, Dictionary<int, object> se
         public void AssertValue(MirrorData data, object value)
         {
             if (data.IsCollection)
+            {
+                if (!(value is IEnumerable))
+                {
+                    Assert.Fail("Data is collection but expected vlaue is not.");
+                }
                 AssertCollection(data, value as IEnumerable);
+            }
             else if (value == null)
+            {
                 Assert.IsTrue(data.IsNull);
+            }
             else if (value is int)
-                Assert.AreEqual((int)value, Convert.ToInt32(data.Data));
+            {
+                try
+                {
+                    int mirrorData = Convert.ToInt32(data.Data);
+                    Assert.AreEqual((int)value, mirrorData);
+                }
+                catch (Exception e)
+                {
+                    Assert.Fail(e.Message);
+                }
+            }
             else if (value is double)
-                Assert.AreEqual((double)value, Convert.ToDouble(data.Data), 0.00001);
+            {
+                try
+                {
+                    double mirrorData = Convert.ToDouble(data.Data);
+                    Assert.AreEqual((double)value, Convert.ToDouble(data.Data), 0.00001);
+                }
+                catch (Exception e)
+                {
+                    Assert.Fail(e.Message);
+                }
+            }
```

Comment:
@zora-wang, `DSEvaluationViewModelUnitTest` has derived classes spanning over multiple test assemblies. You need to run all tests with *Isolation Tool*, which will take around 45 mins. Please set up the run before we leave for lunch, get help from Minh if you need. After the run, we need to know the statistics before this can be merged. Thanks.